echo/tracer.go

```diff
@@ -0,0 +1,166 @@
+package echo
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gookit/color"
+	"github.com/labstack/echo"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	reporter "github.com/wavefronthq/wavefront-opentracing-sdk-go/reporter"
+	"github.com/wavefronthq/wavefront-opentracing-sdk-go/tracer"
+	wfTracer "github.com/wavefronthq/wavefront-opentracing-sdk-go/tracer"
+	wfApplication "github.com/wavefronthq/wavefront-sdk-go/application"
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+)
+
+// Tracer Global Tracer
+var Tracer opentracing.Tracer
+var globalTracerConfig *tracerConfig
+
+// InitTracer initialize tracer object, initialize the GlobalTracer with this newly created,
+// tracer object.
+// Call utils/wavefront.Tracer to access the tracer object
+//For proxy sender send token as empty string
+func InitTracer(cfg Config) error {
+
+	//Constructing globalTracerConfig from config file
+	tracerConfig, err := readConfigFromYaml(cfg.CfgFile)
+	if err != nil {
+		return fmt.Errorf("error reading config file: %w", err)
+	}
+	//Registering routes
+	tracerConfig, err = readConfigFromYaml(cfg.RoutesFile)
+	if err != nil {
+		return fmt.Errorf("error reading routes file: %w", err)
+	}
+
+	var sender senders.Sender
+	if cfg.DirectCfg != nil {
+		// Create the direct sender
+		sender, err = senders.NewDirectSender(cfg.DirectCfg)
+		if err != nil {
+			return err
+		}
+	} else if cfg.ProxyCfg != nil {
+		// Create the proxy sender
+		sender, err = senders.NewProxySender(cfg.ProxyCfg)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New(color.Error.Sprint("Wavefront Middleware... at least one of Direct Sender or Proxy Sender configuration must be passed"))
```

Comment:
nitpick: drop the color dep, and also tweak the log message to start with lower case. Maybe just:
`wavefront middleware: one of Direct sender or Proxy sender configuration is required`.